File: SlidingWindow/maxSubarraySum.js
function maxSubarraySum(arr, num) {
  let max = 0;
  let left = 0;
  let sum = 0;
  if (arr.length < num) {
    return null;
  }
  for (let i = 0; i < num; i++) {
    sum+= arr[i];
  }
  max = sum;
  for (let right = num; right < arr.length; right++) {
    sum = sum - arr[left] + arr[right];
    left++;
    max = Math.max(max, sum);
  }
  return max;
}


console.log(maxSubarraySum([100,200,300,400], 2));

File: SlidingWindow/maxSubarraySum.py
# Online Python compiler (interpreter) to run Python online.
# Write Python 3 code in this online editor and run it.
def maxSubarraySum(arr, num):
    if (len(arr) < num):
        return None
    left = 0
    sumS = 0
    maxS = 0
    for item in range(num):
        sumS = sumS + arr[item]
    maxS = sumS
    
    for item in range(num, len(arr)):
        sumS = sumS - arr[left] + arr[item]
        left = left+1
        maxS = max(maxS, sumS)
    return maxS
    
    
print(maxSubarraySum([100,200,300,400], 2))